code/datums/components/thermo.dm

```diff
@@ -0,0 +1,78 @@
+ // This is basicly a component that acts like a space heater
+ // Like something thatt emits heat like a flare or absorbes heat like ice
+ // You can set up ranges for when it starts smoking or when it catches fire or
+ //  when it ices over and frezes over, but there is no implmention of sever cold yet
+// Most of it is copyed over from the space_heater code
+/datum/component/thermo
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/targetTemperature = T20C + 80
+	var/heatingPower = 10000			// amount of raw heat added, side note, this is the default heat capacity a standard T1 frezzer can put out
+	var/efficiency = 0.25				// amount of air taken from the turf
+	var/temperatureTolerance = 1
+	var/on = TRUE
+	var/current_temp = -1
+
+/datum/component/thermo/Initialize(_targetTemperature = T20C, _heatingPower = 10000, _efficiency = 0.25)
+	targetTemperature = _targetTemperature
+	heatingPower = _heatingPower
+	efficiency = _efficiency
+	START_PROCESSING(SSobj, src)
+
+/datum/component/thermo/UnregisterFromParent()
+	STOP_PROCESSING(SSobj, src)
+
+
```

Comment:
whitespace

Reply:
pretty trivial review considering how many files have odd numbers of newlines, but I'll make a commit for it regardless.